Feat/88 duplicates

f-lukas opened the pull request:
https://github.com/Pandora-IsoMemo/iso-app/issues/88

- A new button "detect duplicates" has been added that opens a pop-up where duplicates can be detected and removed from the dataset.

Note: I have commented out the export button for now, as you can simply use the existing button in the app.

arunge commented on NEWS.md:
```diff
@@ -1,5 +1,10 @@
 # MpiIsoApp development version
 
+## MpiIsoApp 23.01.5
```
Can you use 23.01.4 here? Even if there is another PR in the pipeline: "wer zuerst merged, mahlt zuerst..." ;)

Regarding the other PR, I first want to ensure with Ricardo that the name for the new Repo is fine for him. And only after that, I will merge ...

arunge commented on DESCRIPTION:
```diff
@@ -1,6 +1,6 @@
 Package: MpiIsoApp
 Title: Pandora & IsoMemo spatiotemporal modeling
-Version: 23.01.3
+Version: 23.01.5
```
See below

arunge commented on R/01-createDuplicateModal.R:
```diff
@@ -0,0 +1,131 @@
+                A value of 0 means that the values are rounded to whole numbers.
+                You can also enter negative numbers, which will result in rounding to the power of ten, e.g. the value -2 rounds to the nearest hundred.<br>
+                For text columns you can choose between 'Case Sensitive', where a distinction is made between upper and lower case letters and
+                'Case Insensitive', where no distinction is made.<br><br>
```
Can you make a note after this sentence, that per default for all selected columns the most strict  method for matching is used, e.g. _no rounding_ and _case sensitive_ ?

arunge commented on R/01-createDuplicateModal.R:
```diff
@@ -0,0 +1,131 @@
+    title = "Duplicate Identifier",
+    easyClose = TRUE,
+    size = "l",
+    footer = modalButton("Close (Without Saving Changes)"),
```
Can you shift the button "Close Modal and Transfer to App", see pic.,  and rename it to e.g. "Accept (Transfer to App)"?
![image](https://user-images.githubusercontent.com/16759098/215453270-b2185f94-5994-4aac-8292-dc1acca3fe07.png)

arunge commented on R/01-createDuplicateModal.R:
```diff
@@ -0,0 +1,131 @@
+          4,
+          selectizeInput(
+            inputId = ns("selectedVariable"),
+            label = "Degree of Similarity for Variable...",
```
I think it could help here to have more options in the select input, such as in 
![image](https://user-images.githubusercontent.com/16759098/215454050-039bfc0d-6c54-44c6-8d24-b140dcaabe97.png)

You can compare here: https://github.com/Pandora-IsoMemo/iso-app/blob/61733904472fb6d3449c452b761a0830f7a05a8f/R/03-dataExplorer.R#L156-L159

f-lukas replied:
Great idea, thank you.

arunge commented on R/01-createDuplicateModal.R:
```diff
@@ -0,0 +1,131 @@
+          )
+        )
+      ),
+      checkboxInput(
```
What about shifting the inputs into one row? Than the user would need less scrolling...

![image](https://user-images.githubusercontent.com/16759098/215456224-0e38dced-ea8a-4c07-ad59-4a0ac68b0757.png)

arunge reviewed (changes requested):
In general, I like the new pop up! :+1:  But there are still some issues, e.g.
I have chosen
![image](https://user-images.githubusercontent.com/16759098/215457139-ff1c2d15-7d97-4ab8-8a75-aae6a2c4e5d4.png)
But I have got **not-exact matching duplicates** when showing only duplicates? Why? :thinking: 
![image](https://user-images.githubusercontent.com/16759098/215457600-c6f7bf03-4145-415d-90f7-7f4197dcda64.png)

f-lukas commented:
@arunge Thanks for your feedback. 

I have adjusted a few things. I have now set the version back to 23.01.5 because you have merged the other feature in the meantime.
Thanks for the hint about the bug. Unfortunately I had to remove the highlighting function, although I liked it. With larger data sets, the coloring does not seem to be performant. I get a JS error in the console. Either the table is not displayed at all or not all rows are colored (like in your screenshot). Therefore I have removed the highlighting.

arunge reviewed (dismissed):
Very good! :+1:

arunge commented:
Just merge updates for Description and News.md into this branch before merging. :slightly_smiling_face:
